src/lib/InstantSearch.ts

```diff
@@ -307,13 +337,14 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
    * [open an issue](https://github.com/algolia/instantsearch.js/issues/new?title=Problem%20with%20dispose).
    * @return {undefined} This method does not return anything
    */
-  dispose() {
+  public dispose(): void {
     this.scheduleSearch.cancel();
     this.scheduleRender.cancel();
     clearTimeout(this._searchStalledTimer);
 
     this.removeWidgets(this.mainIndex.getWidgets());
-    this.mainIndex.dispose();
+    // @TODO: do we want this to be correct arguments?
+    this.mainIndex.dispose(({} as any) as DisposeOptions);
```

Comment:
here also, we call this with less than what index asks for

Reply:
We can provide the `Helper` + `state` but we don't use them.